buildSrc/src/main/kotlin/java-library-conventions.gradle.kts

```diff
@@ -201,6 +204,23 @@ afterEvaluate {
 					"-parameters" // Generates metadata for reflection on method parameters.
 			))
 		}
+
+		val modulePath: String  = tasks.compileJava.get().classpath.asPath
+		val compileModuleInfoJava by tasks.registering(JavaCompile::class) {
+			dependsOn(tasks.compileJava)
+
+			destinationDir = tasks.compileJava.get().destinationDir
```

Comment:
I seem to remember that `TaskProvider`s (like `tasks.compileJava`) have a `map` method or equivalent, which I believe could be used here like to following to prevent unwrapping `tasks.compileJava` eagerly:

```gradle
destinationDir = tasks.compileJava.map { it.destinationDir }
```

Does this work for you @sormuras? :)